services/QuillLMS/client/app/bundles/Staff/components/evidence/datasetRecovery/DatasetRecoveryUpload.tsx

```diff
@@ -0,0 +1,224 @@
+    if (!parsedData?.validation.isValid || !parsedData.changes) return;
+
+    sessionStorage.setItem('dataset_recovery_changes', JSON.stringify(parsedData.changes));
+    history.push(parsedData.validation.redirectPath!);
+  };
+
+  const handleClear = () => {
+    setSelectedFile(null);
+    setParsedData(null);
+    setErrorMessage(null);
+    if (fileInputRef.current) {
+      fileInputRef.current.value = '';
+    }
+  };
+
+  const renderSummary = () => {
+    if (!parsedData?.validation.changesSummary) return null;
+
+    const summary = parsedData.validation.changesSummary;
+    const items = [
+      { label: 'Examples to create', value: summary.examplesCreated },
+      { label: 'Examples to update', value: summary.examplesUpdated },
+      { label: 'Examples to delete', value: summary.examplesDeleted },
+      { label: 'Clusters to create', value: summary.clustersCreated },
+      { label: 'Clusters to update', value: summary.clustersUpdated },
+      { label: 'Guidelines to create', value: summary.guidelinesCreated },
+      { label: 'Guidelines to update', value: summary.guidelinesUpdated }
+    ].filter(item => item.value > 0);
+
+    if (items.length === 0) {
+      return <p className="no-changes">No changes found in recovery file</p>;
+    }
+
+    return (
+      <ul className="changes-summary">
+        {items.map(item => (
+          <li key={item.label}>
+            <span className="label">{item.label}:</span>
+            <span className="value">{item.value}</span>
+          </li>
+        ))}
+      </ul>
+    );
+  };
+
+  return (
+    <div className="dataset-recovery-upload">
+      <div className="header">
+        <h1>Dataset Recovery Upload</h1>
+        <p className="description">
+          Upload a recovery JSON file to restore changes that failed to save.
+          The file will be validated and you will be redirected to the appropriate dataset editor.
+        </p>
+      </div>
+
+      <div className="upload-section">
+        <div className="file-input-wrapper">
+          <input
+            accept=".json"
+            onChange={handleFileSelect}
+            ref={fileInputRef}
+            type="file"
+          />
+        </div>
+
+        {isProcessing && (
+          <div className="processing">
+            Processing file...
+          </div>
+        )}
+
+        {errorMessage && (
+          <div className="error-message">
+            {errorMessage}
+          </div>
+        )}
+
+        {parsedData?.validation.isValid && (
+          <div className="validation-success">
+            <h3>File validated successfully</h3>
+            <p className="dataset-id">
+              Target Dataset ID: <strong>{parsedData.validation.datasetId}</strong>
+            </p>
+
+            <h4>Changes to apply:</h4>
+            {renderSummary()}
+
+            <div className="actions">
+              <button
+                className="quill-button-archived contained primary medium"
```

Comment:
NBD but ideally we wouldn't be adding new instances of `quill-button-archived` (this should be roughly equivalent to `quill-button contained medium green focus-on-light`.